repository_service_tuf/cli/artifact/download.py

```diff
@@ -0,0 +1,266 @@
+# SPDX-FileCopyrightText: 2023 VMware Inc
+#
+# SPDX-License-Identifier: MIT
+
+
+import base64
+import os
+from hashlib import sha256
+from pathlib import Path
+from typing import Any, Dict, Optional
+from urllib import request
+
+import yaml
+from click import Context
+from tuf.api.exceptions import DownloadError, RepositoryError
+from tuf.ngclient import Updater
+
+from repository_service_tuf.cli import click, console
+from repository_service_tuf.cli.artifact import artifact
+from repository_service_tuf.cli.artifact.repository import find_hash_bin
+
+DOWNLOAD_DIR = "./downloads"
+
+
+@artifact.command()
+@click.option(
+    "-c",
+    "--config",
+    help="A config file to use.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-r",
+    "--repository",
+    help="A repository to use config for.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-t",
+    "--trusted-root",
+    "trusted_root",
+    help="A metadata URL to the initial trusted root.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-",
+    "--metadata-url",
+    "metadata_url",
+    help="A URL to download TUF metadata from.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-a",
+    "--artifacts-url",
+    "artifacts_url",
+    help="An artifacts base URL to fetch from.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-n",
+    "--arifact-name",
+    "artifact_name",
+    help="An artifact name to get.",
+    type=bool,
+    required=None,
+    default=None,
+)
```

Comment:
I think we don't want it, too many options and it should be an argument.

So the user can simply run

```
rstuf -c rstuf.yaml artifact download my-app.1.0.4.tar.gz
```

```shell
❯ rstuf -c rstuf.yaml artifact download --help
                                                                                                  
 Usage: rstuf artifact download [OPTIONS] ARTIFACT                                                
                                                                                                  
 Downloads artifacts to the TUF metadata using python-tuf ``ngclient``.                           
                                                                                                  
╭─ Options ──────────────────────────────────────────────────────────────────────────────────────╮
│ --repository        -r  TEXT  A repository to use config for.                                  │
│ --trusted-root      -t  TEXT  A metadata URL to the initial trusted root.                      │
│ --metadata-url      -   TEXT  A URL to download TUF metadata from.                             │
│ --artifacts-url     -a  TEXT  An artifacts base URL to fetch from.                             │
│ --hash-prefix       -h        A hash prefix.                                                   │
│ --directory-prefix  -P  TEXT  A prefix for the download dir.                                   │
│ --help                        Show this message and exit.                                      │
╰────────────────────────────────────────────────────────────────────────────────────────────────╯
```

To implement it we add the decorator 
```python
@click.argument('artifact')
@click.pass_context
def download(
    context: Context,
    repository: str,
    trusted_root: str,
    metadata_url: str,
    artifacts_url: str,
    hash_prefix: bool,
    directory_prefix: str,
    artifact,
) -> None:
```